Mykrobe; sonneityping parse_mykrobe_predict changes

awh082834 opened the pull request:
<!--
Thank your for contributing to the Staph-B community!
Please fill in the appropriate checklist below and delete whatever is not relevant.

These recommendations are not meant as a means to keep people from contributing. 
If you need assistance with the following measures, submit your pull request anyway, and we will work with you...
Or you can contact us beforehand via slack or by sumitting a pull request https://github.com/StaPH-B/docker-builds/issues

Documentation on how to contribute can be found at https://staphb.org/docker-builds/contribute/
Documentation for how to create a Dockerfile can be found at https://staph-b.github.io/docker-builds/make_containers
A recommended Dockerfile template can be found at https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile


Please replace all '[ ]' with '[X]' to demonstrate completion.
-->
Hi! 

I would like to put in a PR pertaining to changes made to `sonneityping` in commit [`7d18a7c`](https://github.com/katholt/sonneityping/commit/7d18a7cbfdb1f931765cc0b243a17605b7cff570) used in `mykrobe` v0.12.1. The typo correction and addition of species when `percent_coverage < 90` has solved indexing issues when non sonnei organisms are passed in, however, there hasn't been a release in a few years. I have also added a quick `-h` test to the Dockerfile.

Thank you!

Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/build-files/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number in build-files (i.e. `docker-builds/build-files/spades/3.12.0/Dockerfile`)
   - [ ] (optional) All test files are located in same directory as the Dockerfile (i.e. `build-files/shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `docker-builds/build-files/spades/3.12.0/README.md`)
   - [ ] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

<!-- If this PR is for something else, please add extra descriptions -->

Kincekara commented:
@awh082834 
Hi Andrew,
Thank you for your PR. I will request a few changes before merging your PR.
1. We do not overwrite already published images. So,  create a folder `/buildfiles/mykrobe/0.12.1-sonneityping/` and put your new dockerfile into it. The image will be deployed with the `0.12.1-sonneityping` tag.  Just a reminder, the latest version is 0.13.0. If you want to use the latest version use `0.13.0-sonneityping` where you need.
2. Add a README.md file to that folder. You can find a template at https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/README.md. Alternatively, you can copy from the previous one, but please be sure that you made the necessary changes and add your note for using a specific commit for sonneityping.
3. Please use a newer base image such as `mambaorg/micromamba:2.0.5-ubuntu22.04` to avoid possible security issues.
4. Modify main README file to add your version.
5. Lastly, Dockerfiles are improved over time. It would be good to check out newer ones to avoid possible issues and create a better image. 

I am happy to help if you get any trouble while doing those modifications.

awh082834 commented:
Thank you for the pointers! I have addressed the issues that you cited. The reasoning for sticking with 0.12.1 is that is the image currently in use that utilizes the sonneityping parser needed downstream for our use case and this implementation of the sonneityping commit fixes some bugs that have been observed.

Kincekara commented:
@awh082834 Thank you for your modifications.
Besides cosmetic changes in README, I made a few minor changes in Dockerfile:
- fix labels to pass auto-checks
- force conda package cleaning 
- change "as" casing to remove warnings

It is not required, but if you can add it, it would be helpful to have a test that reflects the bug fix.

awh082834 commented:
@Kincekara Parser test added! As the test reads used by mykrobe are typhi the `out.json` is used as the input and produces the correct characterization of "not S. sonnei" outlining the expected behavior of `7d18a7c`

Kincekara commented:
The tests look good.
```
#13 61.63 mykrobe v0.12.1
#13 62.07 genome	species	final genotype	name	confidence	num QRDR	parC_S80I	gyrA_S83L	gyrA_S83A	gyrA_D87G	gyrA_D87N	gyrA_D87Y	lowest support for genotype marker	poorly supported markers	max support for additional markers	additional markers	node support
#13 62.07 SAMPLE	not S. sonnei	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA					
#13 DONE 62.1s
```

Kincekara commented:
Thank you! I have no more suggestions. I will merge and deploy this PR.

awh082834 commented:
Thank you @Kincekara !

Kincekara commented:
Thank you for your PR!
The images are available on Dockerhub  `staphb/mykrobe:0.12.1-sonneityping` and Quay `quay.io/staphb/mykrobe:0.12.1-sonneityping`
